src/java.base/share/classes/sun/security/pkcs12/PKCS12KeyStore.java

```diff
@@ -1345,7 +1365,7 @@ public KeyStore.Entry engineGetEntry(String alias,
                     return new KeyStore.SecretKeyEntry((SecretKey)key,
                         entry.attributes);
                 }
-            } else if (!engineIsKeyEntry(alias)) {
+            } else if (!internalEngineIsKeyEntry(entry)) {
```

Comment:
```suggestion
            } else {
```

(the matching `if` already checks `internalEngineIsKeyEntry(entry)`)

Reply:
Removed.